R/export_cdisc.R

```diff
@@ -237,103 +308,124 @@ get_subjid <- function(data) {
   }
 }
 
-CDISC_COLS <- list(
-  ADPC = c(
-    "STUDYID",
-    "SUBJID",
-    "USUBJID",
-    "SITEID",
-    "VISITNUM",
-    "VISIT",
-    "AVISITN",
-    "PCSTRESC",
-    "PCSTRESN",
-    "PCSTRESU",
-    "PCORRES",
-    "PCORRESU",
-    "PCTPT",
-    "PCTPTNUM",
-    "ATPT",
-    "ATPTN",
-    "AVAL",
-    "ANL01FL",
-    # Columns taken from the original data if present (still not directly mapped)
-    "SEX",
-    "RACE",
-    "AGE",
-    "AGEU",
-    "AVISIT"
-  ),
-
-  ADPP = c(
-    "STUDYID",
-    "USUBJID",
-    "PPSEQ",
-    "PPGRPID",
-    "PPSPID",
-    "PARAMCD",
-    "PARAM",
-    "PPCAT",
-    "PPSCAT",
-    "PPREASND",
-    "PPSPEC",
-    "PPDTC",
-    "PPSTINT",
-    "PPENINT",
-    "SUBJID",
-    "SITEID",
-    # Columns taken from the original data if present (still not directly mapped)
-    "SEX",
-    "RACE",
-    "AGE",
-    "AGEU",
-    "TRT01P",
-    "TRT01A",
-
-    "AVAL",
-    "AVALC",
-    "AVALU",
-
-    # Not CDISC  ADPP standard
-    "VISIT",
-    "AVISIT"
-  ),
-
-  PP = c(
-    "STUDYID",
-    "DOMAIN",
-    "USUBJID",
-    "PPSEQ",
-    "PPCAT",
-    "PPGRPID",
-    "PPSPID",
-    "PPTESTCD",
-    "PPTEST",
-    "PPSCAT",
-    "PPDTC",
-    "PPORRES",
-    "PPORRESU",
-    "PPSTRESC",
-    "PPSTRESN",
-    "PPSTRESU",
-    "PPSTAT",
-    "PPREASND",
-    "PPSPEC",
-    "PPRFTDTC",
-    "PPSTINT",
-    "PPENINT",
-
-    # Not CDISC PP standard
-    "VISIT",
-    "AVISIT"
-  )
-)
-
-INTERNAL_ANCA_COLS <- c(
-  "exclude", "is.excluded.hl", "volume", "std_route",
-  "duration", "TIME", "IX", "exclude_half.life", "is.included.hl",
-  "conc_groups", "REASON"
-)
+# Helper: adjust class and length for a data.frame based on metadata_variables
+adjust_class_and_length <- function(df, metadata) {
+  for (var in names(df)) {
+    var_specs <- metadata %>% filter(Variable == var, !duplicated(Variable))
+    if (nrow(var_specs) == 0 || all(is.na(df[[var]]))) next
+    if (var_specs$Type %in% c("Char", "text")) {
+      df[[var]] <- substr(as.character(df[[var]]), 0, var_specs$Length)
+    } else if (var_specs$Type %in% c("Num", "integer", "float") &&
+                 !endsWith(var, "DTM")) {
+      df[[var]] <- round(as.numeric(df[[var]]), var_specs$Length)
+    }
```

Comment:
suggestion: A warning for an unexpected input might help catch bugs early:
```suggestion
    } else {
      warning("Unknown var specification type: ", var_specs$Type)
    }
```